server/src/serializers/PuzzleSerializer.js

```diff
@@ -0,0 +1,37 @@
+import BlockSerializer from "./BlockSerializer.js"
+
+class PuzzleSerializer {
+  static async getDetail(puzzle) {
+    const allowedAttributes = ["id", "difficulty"]
+
+    let serializedPuzzle = {}
+    allowedAttributes.forEach(attribute => {
+      serializedPuzzle[attribute] = puzzle[attribute]
+    })
+
+    const boxOne = await puzzle.$relatedQuery("boxone")
+    const boxTwo = await puzzle.$relatedQuery("boxtwo")
+    const boxThree = await puzzle.$relatedQuery("boxthree")
+    const boxFour = await puzzle.$relatedQuery("boxfour")
+    const boxFive = await puzzle.$relatedQuery("boxfive")
+    const boxSix = await puzzle.$relatedQuery("boxsix")
+    const boxSeven = await puzzle.$relatedQuery("boxseven")
+    const boxEight = await puzzle.$relatedQuery("boxeight")
+    const boxNine = await puzzle.$relatedQuery("boxnine")
```

Comment:
okay this is more revealing about the potential ER diagram. 

Looks like a Puzzle will have-one related BoxOne record, then one BoxTwo, etc. Each Box has columns corresponding to specific squares. 

While this will work for preseeded Boxes/Puzzles, the difficulty is likely going to arise when you allow for users to be able to create new puzzles, or when you create game logic that potentially checks to ensure that a square value entry doesnt conflict with a number in the same row or column as that square in the grander puzzle. In order to create that logic, you will likely need a significant amount of "if...else" logic e.g. 

"The user just placed a "2" in BoxOne, squareTwo. I'll have to have logic that checks all of the squareTwo, squareSix and squareEight squares in BoxFour and BoxSeven. I'll also have to check all squareOne, SquareTwo and squareThree boxes in BoxOne, BoxTwo and BoxThree, as well as in all squares of BoxOne."
Thats not impossible, but its going to take a LOT of logic. The algorithm/function or even javascript objected needed could get pretty meaty. 

Alternatively, you could store all of your entries as "rows and columns" of a compound array data structure stored in ONE column of your database like we discussed. This might make the logic needed for "checking" for correct entires much easier, and will likely make creating new Puzzles much easier as well. Regardless, it will still require a decent amount of logic. 
````
[ [1, 3, 2, 4, 6, 5, 7, 8, 9] [2, 8...], [...]]
````
or turned out 
```
[
  [ 1, 3, 2, 4, 6, 5, 7, 8, 9 ],
 [ 1, 3, 2, 4, 6, 5, 7, 8, 9 ],
[ 1, 3, 2, 4, 6, 5, 7, 8, 9 ],
[ 1, 3, 2, 4, 6, 5, 7, 8, 9 ],
[ 1, 3, 2, 4, 6, 5, 7, 8, 9 ],
[ 1, 3, 2, 4, 6, 5, 7, 8, 9 ],
[ 1, 3, 2, 4, 6, 5, 7, 8, 9 ],
[ 1, 3, 2, 4, 6, 5, 7, 8, 9 ],
[ 1, 3, 2, 4, 6, 5, 7, 8, 9 ]
]
```

That said, your app can be much simpler in terms of logic with less checking and unvalidated puzzle creation (you put the onus on the user, and hope they create their puzzle correctly)

So if you like, keep going, but know that this is a crossroads in your application. The further you go down one path, the more time you will have to take to double back if you decide to switch course. We can chat more in person!